### ISSUE TITLE
dataset structure for custom dataset

### ISSUE THREAD

**Issue by saifhassan** [IPMI-ICNS-UKE/sparsam#2]
Hey,

It's a great contribution, I want to implement your code on image classification, Can you guide regarding structure of dataset? and what else I will have to do changes?

I have gone through config files but didn't understand, can you please specify the structure. 

Thanks in advance.

**Comment by Moxnie** [comment]
In the Dataset file you find a DatasetBaseclass and the ImageSet class we are using. In many cases you might be able to just use our Class, which requires a list of the corresponding images and in case of the labeled_train_set/ val_set labels. You may also provide the image_size for resizing and a list class_names for correct label mapping. 
Please set the normalize flag to True for anything but the unlabeled_train_set (in this case normalization is performed after DataAugmentation).
If you choose to implement your custom data_set it must return the image as the first argument and a label (could be dummy, for unlabeld data) as a second.

**Comment by saifhassan** [comment]
> In the Dataset file you find a DatasetBaseclass and the ImageSet class we are using. In many cases you might be able to just use our Class, which requires a list of the corresponding images and in case of the labeled_train_set/ val_set labels. You may also provide the image_size for resizing and a list class_names for correct label mapping.
> Please set the normalize flag to True for anything but the unlabeled_train_set (in this case normalization is performed after DataAugmentation).
> If you choose to implement your custom data_set it must return the image as the first argument and a label (could be dummy, for unlabeld data) as a second.

can you guide me what should be the structure of dataset? as I could not understand config files ?

**Comment by Moxnie** [comment]
I just pushed an updated version of the BaseSet, you can just inherit from it and implement your own "_get_image_label_pair" function, which gets an index and should return an image - label pair

**Comment by saifhassan** [comment]
> I just pushed an updated version of the BaseSet, you can just inherit from it and implement your own "_get_image_label_pair" function, which gets an index and should return an image - label pair

Thank you very much for your reply.

I am bit confused about dataset, I have image classification dataset in following structure and I want to implement your code, can you guide me regarding the changes overall in your code:

```
train
       - label1
            - img1.jpeg
            - img2.jpeg
            - img3.jpeg
      - label2
            - img1.jpeg
            - img2.jpeg
            - img3.jpeg
test
       - label1
            - img1.jpeg
            - img2.jpeg
            - img3.jpeg
      - label2
            - img1.jpeg
            - img2.jpeg
            - img3.jpeg
```

**Comment by Moxnie** [comment]
you need to lists the first one containing the paths to your images: 
[root/folder/train/label1/img1.jpeg, ..., root/folder/train/label2/img3.jpeg] 
and a second one containing the corresponding labels:
[label1, ..., label2]

I hope this helps

**Comment by saifhassan** [comment]
> you need to lists the first one containing the paths to your images: [root/folder/train/label1/img1.jpeg, ..., root/folder/train/label2/img3.jpeg] and a second one containing the corresponding labels: [label1, ..., label2]
> 
> I hope this helps

Sorry for inconvenience, but I donot understand your solution. My query is:

I have dataset in structure above mentioned, now I want to run your `dino_example.py` code, so overall what and where the changes are required? Please mention filename where changes required and what changes in that file.

It'll be great help if you do this.

Thanks

**Comment by Moxnie** [comment]
The code in example.py is only an abstract example, therefore you need to define your own datasets. To do so you have two options.
1.: Define you own Dataset, which inherits from `BaseSet` in `utils,DataSet`
2.:  You may use the `ImageSet` class in `utils.DataSet`. Then you need to extract the paths to your images and labels as mentioned above.
Afterwards replace the `BasSet()` in line 22  with Option 1 or 2 and initialize accordingly.
Repeat in lines 26, 27 and 29 and set your desired DataLoader parameter.
Further you may choose your own backbone in line 32 e.g. ` timm.models.xcit_small_12_p8_224_dist`. 
Set data_loader_parameter in line 36 e.g. batch_size (would recomend 256) etc.
Anything else is optional, further information are found after the "# TODO: "
Finally use Option 1 and remove line 81 to 91.
